engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/kafka/generic/KafkaDelayedSourceFactory.scala

```diff
@@ -0,0 +1,44 @@
+package pl.touk.nussknacker.engine.kafka.generic
+
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.{CustomNodeError, NodeId}
+import pl.touk.nussknacker.engine.api.definition.{DualParameterEditor, MaximalNumberValidator, MinimalNumberValidator, Parameter, StringParameterEditor}
+import pl.touk.nussknacker.engine.api.editor.DualEditorMode
+import pl.touk.nussknacker.engine.api.typed.typing.Typed
+
+object KafkaDelayedSourceFactory {
+
+  import scala.collection.JavaConverters._
+
+  val delayValidators = List(MinimalNumberValidator(0), MaximalNumberValidator(Long.MaxValue))
+
+  final val DelayParameterName = "delayInMillis"
+
+  final val DelayParameter = Parameter.optional(DelayParameterName, Typed[java.lang.Long])
+
+  final val TimestampFieldParamName = "timestampField"
+
+  // TODO: consider changing to lazy parameter
```

Comment:
Can you extend this comment to "and add the same parameter also in "not delayed" kafka sources?

Reply:
Comment added.